File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/EditTask_RetryParams.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

import com.google.gson.annotations.SerializedName;

data class EditTask_RetryParams(
  val id: FixupTaskID,
  val instruction: String,
  val model: String,
  val mode: ModeEnum, // Oneof: edit, insert
  val range: Range,
) {

  enum class ModeEnum {
    @SerializedName("edit") Edit,
    @SerializedName("insert") Insert,
  }
}

File: agent/bindings/csharp/EditTaskRetryParams.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class EditTaskRetryParams
  {
    [JsonProperty(PropertyName = "id")]
    public FixupTaskID Id { get; set; }
    [JsonProperty(PropertyName = "instruction")]
    public string Instruction { get; set; }
    [JsonProperty(PropertyName = "model")]
    public string Model { get; set; }
    [JsonProperty(PropertyName = "mode")]
    public ModeEnum Mode { get; set; } // Oneof: edit, insert
    [JsonProperty(PropertyName = "range")]
    public Range Range { get; set; }

    public enum ModeEnum
    {
      [EnumMember(Value = "edit")]
      Edit,
      [EnumMember(Value = "insert")]
      Insert,
    }
  }
}
